homeassistant/components/roborock/roborock_storage.py

```diff
@@ -0,0 +1,152 @@
+"""Roborock storage."""
+
+import dataclasses
+import logging
+import os
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.storage import Store
+from homeassistant.util import dt as dt_util
+
+from .const import DOMAIN
+from .models import RoborockMapInfo
+
+STORAGE_VERSION = 1
+_LOGGER = logging.getLogger(__name__)
+MAP_PATH = f"{DOMAIN}/maps"
+MAP_UPDATE_FREQUENCY = 3600  # Only save the map once every hour.
+
+
+@dataclasses.dataclass
+class RoborockMapEntry:
+    """Describe a map entry stored on disk."""
+
+    map_name: str
+    time: float
+
+
+class RoborockStorage:
+    """Store Roborock data."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        entry_id: str,
+    ) -> None:
+        """Initialize RoborockStorage."""
+        self._hass = hass
+        self._data: dict[str, RoborockMapEntry] = {}
+        self._entry_id = entry_id
+        self._store: Store = Store(hass, STORAGE_VERSION, f"{DOMAIN}.{self._entry_id}")
+        self._map_info: dict[str, dict[int, RoborockMapInfo]] = {}
+
+    def _should_update(self, map_entry: RoborockMapEntry | None) -> bool:
+        return (
+            map_entry is None
+            or dt_util.utcnow().timestamp() - map_entry.time > MAP_UPDATE_FREQUENCY
+        )
+
+    def _get_map_filename(self, map_name: str) -> str:
+        return self._hass.config.path(f"{MAP_PATH}/{self._entry_id}/{map_name}")
+
+    def exec_load_maps(self, map_names: list[str]) -> list[bytes | None]:
+        """Load map content. Should be called in executor thread."""
+        filenames: list[tuple[str, str]] = [
+            (map_name, self._get_map_filename(map_name)) for map_name in map_names
+        ]
+
+        results: list[bytes | None] = []
+        for map_name, filename in filenames:
+            try:
+                if not os.path.exists(filename):
```

Comment:
it does not look like `exists` will raise an `OSError` so I think it can be pulled out of the try/except